lib/std/os.zig

```diff
@@ -3327,3 +3327,106 @@ pub fn getrusage(who: i32) rusage {
         else => unreachable,
     }
 }
+
+pub const termios = extern struct {
+    iflag: tcflag_t,
+    oflag: tcflag_t,
+    cflag: tcflag_t,
+    lflag: tcflag_t,
+    line: cc_t,
+    cc: [NCCS]cc_t,
+    ispeed: speed_t,
+    ospeed: speed_t,
+
+    pub const TermiosError = error {
+        SyscallFailed,
+        UnknownAction,
+    };
+
+    pub fn get(handle: fd_t) TermiosError!@This() {
+        if (builtin.os == .linux) {
+            var term: @This() = undefined;
+            const err = linux.syscall3(linux.SYS_ioctl, @bitCast(usize, @as(isize, handle)), linux.TCGETS, @ptrToInt(&term));
+            if (err != 0) {
+                return TermiosError.SyscallFailed;
+            }
+            return term;
+        } else {
+            @compileError("termios unimplemented for this target");
+        }
+    }
+
+    pub fn set(self: *@This(), handle: fd_t, optional_action: isize) TermiosError!void {
+        if (builtin.os == .linux) {
+            if (optional_action < @This().tcsanow or optional_action > @This().tcsaflush) {
+                return TermiosError.UnknownAction;
+            }
+
+            const err = linux.syscall3(linux.SYS_ioctl, @bitCast(usize, @as(isize, handle)), linux.TCSETS + @bitCast(usize, optional_action), @ptrToInt(self));
+            if (err != 0) {
+                return TermiosError.SyscallFailed;
+            }
+        } else {
+            @compileError("termios unimplemented for this target");
+        }
+    }
+
+    // used for set()
+    pub const tcsanow = TCSANOW;
+    pub const tcsadrain = TCSADRAIN;
+    pub const tcsaflush = TCSAFLUSH;
+
+    // iflag bits
+    pub const IGNBRK = 1;
```

Comment:
Are these values *always* the same across platforms?

Reply:
If you're talking about the flag bits values, then yes, I've checked.

Reply:
When I say platforms I mean operating systems: i.e. are these values specified by POSIX?
And are they the same on OSX and BSDs?

Reply:
Yes, these values are defined in the [POSIX standard](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap11.html).

Reply:
The define names are, but the values are not as far as I can see?

Reply:
After rechecking again, you're right; some of the values do differ. But that's exactly why I've replaced these values to be inside `bits/linux.zig` now.

If any *BSD users want to implement this, they know where the proper place to put these values is.